honeybee_energy/properties/shade.py

```diff
@@ -51,42 +39,31 @@ def host(self):
         return self._host
 
     @property
-    def diffuse_reflectance(self):
-        """Get or set the diffuse reflectance of the shade."""
-        return self._diffuse_reflectance
-
-    @diffuse_reflectance.setter
-    def diffuse_reflectance(self, value):
-        self._diffuse_reflectance = float_in_range(
-            value, 0, 1, 'shade diffuse reflectance')
-        assert self._diffuse_reflectance + self._specular_reflectance <= 1, \
-            'Sum of diffuse and specular reflectance is greater than 1. Got ' \
-            '{}'.format(self._diffuse_reflectance + self._specular_reflectance)
-
-    @property
-    def specular_reflectance(self):
-        """Get or set the specular reflectance of the shade."""
-        return self._specular_reflectance
-
-    @specular_reflectance.setter
-    def specular_reflectance(self, value):
-        self._specular_reflectance = float_in_range(
-            value, 0, 1, 'shade specular reflectance')
-        assert self._diffuse_reflectance + self._specular_reflectance <= 1, \
-            'Sum of diffuse and specular reflectance is greater than 1. Got ' \
-            '{}'.format(self._diffuse_reflectance + self._specular_reflectance)
-
-    @property
-    def transmittance(self):
-        """Get or set the transmittance of the shade."""
-        return self._transmittance
-
-    @transmittance.setter
-    def transmittance(self, value):
-        assert self._transmittance_schedule is None, \
-            'Shade transmittance cannot be be set while a transmittance ' \
-            'schedule is assigned.'
-        self._transmittance = float_in_range(value, 0, 1, 'shade transmittance')
+    def construction(self):
+        """Get or set a ShadeConstruction for the shade."""
+        if self._construction:  # set by user
+            return self._construction
+        elif not self._host.has_parent:
+            return generic_context
+        elif hasattr(self._host.parent.properties.energy, 'construction_set'):
```

Comment:
The next 3 checks will be probably cleaner if we write them as a single recursive method but this will also work.

Reply:
Excellent suggestion!  Implemented!